### ISSUE TITLE
cdr-atom when called with a list of variables

### ISSUE THREAD

**Issue by noskill** [trueagi-io/hyperon-experimental#488]
!(cdr-atom ($A $B $C))

**Comment by vsbogd** [comment]
@noskill , please explain what do you expect, what do you see and which version of the interpreter do you use: Rust or minimal MeTTa?

**Comment by Necr0x0Der** [comment]
I dug into the issue a little bit deeper. The original problem was that
```
(: makelist (-> Atom Atom))
(= (makelist $x)
   (if (== () $x) Nil
       (let $cdr (cdr-atom $x) (Cons (car-atom $x) (makelist $cdr)))))

! (makelist ($A $B $C))
```
doesn't work even after #473 . I encountered that `!(cdr-atom ($A $B $C))` doesn't work returning `(Error (cdr-atom $C) cdr-atom expects one argument: expression)`, which turned out to be confusing, because it looked like `!(cdr-atom ($A $B $C))` is incorrectly reduced to `(cdr-atom $C)`. But it appeared that this behavior depends on the content of the space that implies that some matching takes place.
Basically, it appeared that #473 doesn't stop reduction of variable atoms by default (as it was mentioned somewhere). If you remove `"variable_operation"` from `default = ` in `Cargo.toml`, the code above starts to work. @noskill , take a note.

**Comment by noskill** [comment]
@Necr0x0Der does it work on minimal metta?

**Comment by noskill** [comment]
@vsbogd I expect !(cdr-atom ($A $B $C)) to return ($B $C), trying it on minimal metta

**Comment by noskill** [comment]
@Necr0x0Der when i run with python runner process never stops, when i run with rust runner  it prints 
```
[(Cons $A (Cons Cons (Cons (Error (car-atom $C) car-atom expects one argument: expression) (Cons (Cons Error (Cons (Error (cdr-atom $C) cdr-atom expects one argument: expression) (Cons cdr-atom expects one argument: expression Nil))) Nil))))]
```

**Comment by Necr0x0Der** [comment]
Yes, it seems to not work on Minimal MeTTa, so we can reopen it specifically for this.
